### PR TITLE
Use bundled zlib on all versions, not just 21

### PR DESCRIPTION
**Pull request by sxa**
Fixes https://github.com/adoptium/temurin-build/issues/1951

### CODE REVIEW

**Review by karianna — APPROVED** [review]

**Review by steelhead31 — APPROVED** [review]
LGTM!

**Review by andrew-m-leonard — APPROVED** [review]

File: sbin/build.sh [hunk 1]
```diff
@@ -548,9 +548,7 @@ configureFreetypeLocation() {
 
 configureZlibLocation() {
   if [[ ! "${CONFIGURE_ARGS}" =~ "--with-zlib" ]]; then
-    if [[ "${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}" -ge 21 ]]; then
         addConfigureArg "--with-zlib=" "bundled"
```
Comment (tellison) [inline sbin/build.sh:551]: Fix indentation only.

```suggestion
    addConfigureArg "--with-zlib=" "bundled"
```

**Review by karianna — APPROVED** [review]

**Comment by AdamBrousseau** [comment]
@sxa could this be wrapped in an `if temurin`? I added it on my side because OpenJ9/Semeru does not want to bundle zlip currently.

**Comment by tellison** [comment]
@AdamBrousseau sure. Please send a PR for that (although maybe a !OpenJ9 test rather than =Temurin to preserve the current behaviour for other users)

**Pull request merged**